README reworked (#16)
kitsunyan/foxy-droid: README.md

## Description

Unofficial F-Droid client in the style of the classic one.

> Jump over the lazy dog, manage repositories and install software quickly.

Reviewer: Shouldn't there be a comma before "and"?

Author: Nice catch!

> Jump over the lazy dog, manage repositories and install software quickly.

Author: ```suggestion
Jump over the lazy dog, manage repositories, and install software quickly.
```

No privileged extension, root installation, or sharing local repositories nearby.

### Features

* Classic F-Droid style.
* No cards or inappropriate animations.
* Fast repository syncing.
* Standard Android components and minimal dependencies.

> * Attention to detail in programming logic and design.

Reviewer: "programming logic" concerns me here. Initially it was "program logic" like "logic in program code", not like "logic in programming". I think "program logic" is more correct.

Reviewer: Actually, now I think that it would be better to remove this line at all.

## Description

> Unofficial F-Droid client resembling the classic F-Droid client.

Author: ```suggestion
Unofficial F-Droid client in the style of the classic one.
```
Could "Unofficial" be replaced with "Alternative". The F-Droid folks are all about alternative clients.

Jump over the lazy dog, manage repositories and install software quickly.

> No privileged extension, root installation, or sharing local repositories nearby yet.

Author: ```suggestion
No privileged extension, root installation, or sharing local repositories nearby.
```
Maybe with https://f-droid.org/en/packages/com.genonbeta.TrebleShot/ sometime in the future.

## Features

* Classic F-Droid style.
* No cards or inappropriate animations.
* All apps shown in "Available".
* Fast repository syncing.
* Standard Android components and minimal dependencies.

> * Attention to detail in programming logic and design.

Reviewer: "detail**s**"

Reviewer: Actually, I may be not right, seems "attention to detail" is a correct way too. But since I mention "program login _and_ design", detail**s** may be a better option. Or not.

Author: You could be, but in my head that is "Attention to the details of ___."
"Attention to detail" is a figure of speech, so I think it is correct.
"Pedantry in" or "Pedantic __" should be a better option, but isn't.

Reviewer: >  "Attention to detail" is a figure of speech, so I think it is correct.

Yes, I know (tbh, didn't know about it when opened this review, but then searched for it). I wanted to suggest another option since for me it looks more clean (again, it's my sense of English: if I translated this line to Russian, "detail" will be translated to "details").

Closing this.

## Description

> Unofficial F-Droid client resembling the classic F-Droid client.

Reviewer: I'm not quite sure it's correct to replace it so. I'm not a native English speaker, but is sounds weird to me.

Author: Transitive verbs are just a neater way to relate things to each-other.
https://www.dictionary.com/browse/resembling

To illustrate a sentence that is a logic extreme of tediousness:

"The client of F-Droid that is unofficial is the one that resembles the client of F-Droid that is classic."

Reviewer: > Transitive verbs are just a neater way to relate things to each-other.

Probably, but that's because "resembling" is not a common word. It can be replaced to something more common, I don't feel like "resembling" should be left here (though, I don't mind "that resembles", it feels normal to me).

Author: "That resembles" is not optimal, and not a common word. Will revise.

Jump over the lazy dog, manage repositories and install software quickly.

> No privileged extension, root installation, or sharing local repositories nearby yet.

Reviewer: Didn't get the "Jump over the lazy dog" first, but it's actually funny :) Probably it's a good idea to rework it this way.

Reviewer: The only problem here is "nearby yet", which implies these features may be available in the future. "local repositories" is something that will never be available in Foxy Droid. And I don't think it's right to leave assumptions that these features may be available in the future.

> ## Features

Reviewer: It was intentional to use 3rd-level header here.

Author: I know, but the overall appearance I found unsymmetrical and noisy. I value the orderliness over the information that some things belong to each-other. That information isn't carried over well, it just appears different enough that it looks off. So it becomes a tool to remember what belongs where to the writer, which isn't needed.

Reviewer: > I found unsymmetrical and noisy

I don't think so, and I prefer the old structure, this is how I structure my READMEs. Please, revert it (and on "Screenshots" block too).

Author: Will do.

* Classic F-Droid style.
* No cards or inappropriate animations.

> * All apps shown in "Available".

Reviewer: I think "classic F-Droid style" implies that. This point is not necessary.

Author: Which one? People might not know what "Classic F-Droid style" is.

Reviewer: They I don't think it should be described this way.

"A single-listed catalog of applications" or something like that is better to describe that, instead of pointing out the "Available" tab.

Author: "Uninterrupted listing of apps" maybe. G-Droid made a big point of this. Removed.

> * Fast repository syncing.

Reviewer: I though it's actually important to notice about "network conditions". F-Droid servers like to hang sometimes, so everything is slow as hell, and it's not application-dependent.

Author: Then throw an error that suggests the culprit is F-Droid upstream.
If the user wants to decide whether to use Foxy Droid, things like this are just superfluous.
F-Droid itself doesn't mention it, and the user isn't getting away from it between the two.
Nobody is going to hold it against Foxy Droid that this was not mentioned. It is a fatalist sentence that does not belong in the prominent features.

Reviewer: Ok.

* Standard Android components and minimal dependencies.
* Attention to detail in programming logic and design.

> * New-and-shiny, _and_ "old one was better".

Reviewer: I also think that this point is not necessary.

Author: ```suggestion
```
Was a joke, will remove. The idea was that nobody can complain, as those are the two qualities people go for, and here they meld.

> ## Screenshots

Reviewer: (and here too)

## Building and Installing

> Specify your Android SDK path either using the `ANDROID_HOME` environment variable, or filling out the `sdk.dir` property in `local.properties`.

Author: ```suggestion
To join the development, specify your Android SDK path either using the `ANDROID_HOME` environment variable, or by filling out the `sdk.dir` property in `local.properties`.
```
How about this, the idea is to allow those that don't need it to discard the info without being scared of it.

> To join the development, specify your Android SDK path either using the `ANDROID_HOME` environment variable, or by filling out the `sdk.dir` property in `local.properties`.

Reviewer: "To join the development" is not correct. What should they do if they don't want to join the development, but just want to build it?

The purpose of this block is to describe how to build and install the package, that's all. It's important because GPL actually requires such explanation, so it's placed into readme.

As I said last time reviewing this block, I prefer my version, because this part is purely technical, and shouldn't imply anything.

Reviewer: Also, I limit all lines in this file to 120 characters.

> To join the development, specify your Android SDK path either using the `ANDROID_HOME` environment variable, or by filling out the `sdk.dir` property in `local.properties`.

Author: ```suggestion
To join the development, specify your Android SDK path either using the `ANDROID_HOME` environment variable,
or by filling out the `sdk.dir` property in `local.properties`.
```

## Building and Installing

To join the development, specify your Android SDK path either using the `ANDROID_HOME` environment variable,

> or by filling out the `sdk.dir` property in `local.properties`.

Reviewer: > "To join the development" is not correct. What should they do if they don't want to join the development, but just want to build it?

> The purpose of this block is to describe how to build and install the package, that's all. It's important because GPL actually requires such explanation, so it's placed into readme.

> As I said last time reviewing this block, I prefer my version, because this part is purely technical, and shouldn't imply anything.

Author: And conversely, should people not wanting to join the development remove any property in "sdk.dir"? ;)
The idea was to not scare off those that don't know. I'll think of a combination.

Reviewer: But section is named "building and installing", so this section is definitely for, let's say, __hackers__. This is why I want to keep it this way.

> I actually prefer my readme structure. It's more serious, or something like that.

You're suggesting to change it to make it more "user-friendly" and interesting, and I can agree that it's the case for "Description" block (as well as its "Features" and "Screenshots" subblocks) (And they will also be displayed on F-Droid application page). But it's not the case for the rest text.

Author: Entirely correct, I have corrected it.

Reviewer: Also, are you sure comma is necessary before "or by filling out"?

Author: @kitsunyan It is a natural pause. Without the comma it is more of a nor-gate logic where both together are good.

Reviewer: > It is a natural pause

Hm, ok.

## Building and Installing

> Specify your Android SDK path either using the `ANDROID_HOME` environment variable, or filling out the `sdk.dir` property in `local.properties`.

Reviewer: This time I prefer my version.

key.password=key-password
```

Run `./gradlew assembleRelease` to build the package, which can be installed using the Android package manager.

## License

> GPLv3+ copylefted libre software. Copyright © 2020 kitsunyan.

Reviewer: (wtf, where my original comment has gone)

I'd preferred to leave it as it is. I use the same line for my other project. It seems that it's also better to use (C) instead of ©, especially since it's the only place where copyright information is mentioned (I didn't add headers into the code).

Author: A link to the source code would do nicely here. (C) doesn't actually mean anything, from what I can remember. Headers? © is just UTF-8.
Arguably "Copyright © 2020 kitsunyan." isn't strictly needed either (I think), but having it isn't wrong.

Reviewer: > isn't strictly needed either

Actually, yes. I don't mind removing it. But I want to keep "Foxy Droid is available under the terms of GNU General Public License v3 or later."

Reviewer: > (C) doesn't actually mean anything, from what I can remember. 

I could confuse them. This is why I want to remove "Copyright" information.

> Headers? © is just UTF-8.

I mean, it's common to place a header into each file of the project with legal information. I don't want to do this.

Author: > Foxy Droid

already know what the program is.

>is available under the terms of

this sounds like the availability is limited, and it sounds like the terms are going to be limiting its availability, when one could argue the opposite is true.

> GNU General Public License v3 or later.

there should be a "the" here, and a comma. v3 matches GPL. "version" matches "GNU General Public License", similar to + vs "later".
The only piece of valuable information is that GPLv3 is a license.
"GPLv3+ licensed, copylefted libre software" would suffice.

Reviewer: > already know what the program is.

Yes, think about this line like about something said by a lawyer :) (I'm not a lawyer ofc). So it's okay to leave it here.

> this sounds like the availability is limited

Yes, it's my right since I'm a copyright holder :> So I'm making it available only under GPL v3 or any later version. I don't make it available under any EULA, Apache License, MIT License, or anything else. Only GPL v3 or any later! As I said, this like is more "strict", especially since it's the only place I'm mentioning copyright information across the whole project repository.

> there should be a "the" here

Probably, yes. Even on gnu.org they use "the". Can be changed.

> and a comma

Comma is definitely not necessary (since there are only 2 options, "v3 or later"; comma is used when there are 3 and more options, like "v2, v3, or later", just an example).

Author: What I meant is that the availability is ensured with GPLv3+, and it should ideally sound like it for people that don't know. Oxford comma makes the second option immediately distinct, though not needed here, agreed.
"Copyleft" is a copyright concept. It is already clear this is your copyrighted material from the name being next to "copyright", and the symbol for it. Spending a sentence on what the libre software+ copyleft freedoms entail is better spent. "Use, see, change and share at will, with all." or some-such works. I am not sure about the intention here, but legal matters are maintained. The important part is presenting the project.

Reviewer: > availability is ensured with GPLv3+

That's not quite right. License it something like a "contract": I'm giving a copy of the program under some conditions. So when I say "available under GPL", that means "by using this program you're agreeing with the terms of GPL v3 or any later". I don't provide the choice of another "contracts", only listed one. What you're saying about "availability" is ensured by copyleft _after_ user gets a copy.

But what is really ensured by GPL is that the software is "libre" and "copylefted". In my opinion, your "copylefted libre software" feels kinda "buzzwordy" to me (I noticed you like to place "copylefted libre software" to other projects) (not like I'm trying to shame you or something like that), which is diametrically opposed to my vision of this line to be "strict". There are other words which can describe the program, not only "libre" and "copylefted": "free from patent prosecution", "tivoization-prohibited", etc, etc. But for some reason you want to mention only "libre" and "copylefted", which makes it even more "buzzwordy".

Ultimately, it's about personal preference: make this line "interesting and joyful" or "boring and strict". I explained why I want the second option. I also think, when it comes to something about "legal", it's more important to explain restrictions, than freedoms, so no one could be fooled.

Although, it's a good idea to insert "the" and replace copyright symbol.

Author: From the point of view of the user, copyleft is a premise of availability. I agree my explanation didn't take into account the specifics of how. "libre" and "copyleft" are the only points made, and those are the only ones qualifying it over not being so. Libre is chosen because it is unencumbered with the FSF and the OSI, and it is copyleft as an addition to that. It carries nothing but qualifying statements, and it seems to work a lot better than FSF language. "Full" and "strong" copyleft is more specifically what it is. The patent litigation point is a good one to make.
It can be welcoming and strict. Boring and unwelcoming is what I want to get away from. Made a point to explain it in a way that makes a point of the copyleft requirement.

> GPLv3+ copylefted libre software. Copyright © 2020 kitsunyan.

Author: ```suggestion
GPLv3+ licensed, copylefted libre software. Copyright © 2020 kitsunyan.
```
